app.rb

```diff
@@ -25,10 +26,18 @@ class OpenidConnectRelyingParty < Sinatra::Base
   end
 
   get '/auth/result' do
-    token_response = token(params[:code])
-    userinfo_response = userinfo(token_response[:id_token])
+    code = params[:code]
 
-    erb :success, locals: { userinfo: userinfo_response }
+    if code
+      token_response = token(code)
+      userinfo_response = userinfo(token_response[:id_token])
+
+      erb :success, locals: { userinfo: userinfo_response }
+    else
+      error = params[:error] || 'missing callback param code'
```

Comment:
it seems we don't have test coverage for the case of `'missing callback param code'` - should we?